bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "''",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "paired and nested with non-bracket chars",
+      "input": "([{f00: bar}({}[baz])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested with non-bracket chars",
+      "input": "([{this: sh0ld({}[be] false)])",
```

Comment:
I think adding numbers to test would be worthwhile to do.  But those numbers may look more like expressions you may see like "((5 -3) * 12)" or perhaps "(((3*(12.223+ 15)) - 7)*21)/7" 

Or maybe a statement such as:

```latex
latexM =
\left(\begin{array}{cc} \frac{1}{3} & x\\ \mathrm{e}^{x} &...
 x^2 \end{array}\right)
```

Nothing that is tricky due to the shape of a character compared to another character, visually, just a specific set of patterns you are testing for that may be an edge case for a problem being solved.

The latex statement shows that slashes do not need to be matched, but brackets do, noting that there are an odd number of slashes in that statement.  The first example showing that forward slashes also do not need to be matched.

Those have numbers, look like something you would write, and could be broken down to individual tests specifically addressing the incidental presence of numbers.